### DISCUSSION TITLE
class_namespace accepts an array

### DISCUSSION THREAD

**Discussion by joshbruce** [livewire/livewire#1984] [3. Feature ideas]
I think this is a new idea that's not currently possible.

I have one Laravel install that effectively manages multiple sites using service providers. This basically allows me to keep a separation between app-specific code and Laravel. 

Near as I can tell from the documented example, I can change the default namespace (app/Http/Livewire) to one (and only one) of my providers.

In other words I couldn't given array of namespaces, run the Livewire discover command and have it just register across all the providers. 

It seems this namespace is also tied to the make function. If there was more than one namespace there could be a list response prompt, which I think would be less error prone than someone trying to remember and type it correctly or copying and pasting.

**Comment by Wulfheart** [comment]
Related t0 https://github.com/livewire/livewire/discussions/4496

**Comment by joserick** [comment]
I have had the same problem so I have created a package which allows you to load components from different namespaces, note that this is only for livewire 3.

[Laravel Livewire Discover - https://github.com/joserick/laravel-livewire-discover](https://github.com/joserick/laravel-livewire-discover)

``` php
public function boot(): void
{
	// Load multiples namespace for Livewire components.
	Livewire::componentNamespace('my-components', 'Namespace\\Livewire');
	Livewire::componentNamespace('new-components', 'User\\Repository\\Livewire');
	...
}
```
``` html
<livewire:my-components.devices />
<livewire:new-components.devices-table />
```
